src/pages/SignUp/SignUp.js

```diff
@@ -0,0 +1,252 @@
+        } else if (value.length > 3) {
+          fieldErrors.age = '나이는 세 자리까지 입력해 주세요.';
+        } else {
+          fieldErrors.age = '';
+        }
+        break;
+      case 'height':
+        if (1 > value) {
+          fieldErrors.height = '키를 입력하세요.';
+        } else if (value > 300) {
+          fieldErrors.height = '300이하로 입력해 주세요.';
+        } else {
+          fieldErrors.height = '';
+        }
+        break;
+      case 'seletalMuscleMass':
+        if (1 > value) {
+          fieldErrors.skeletalMuscleMass = '골격근량을 입력하세요.';
+        } else if (value > 100)
+          fieldErrors.skeletalMuscleMass =
+            '골격근량은 100이하로 작성해 주세요.';
+        else {
+          fieldErrors.skeletalMuscleMass = '';
+        }
+        break;
+      case 'goalWeight':
+        if (1 > value) {
+          fieldErrors.goalWeight = '목표 체중을 입력하세요.';
+        } else if (value > 500)
+          fieldErrors.goalWeight = '목표 체중은 500이하로 작성해 주세요.';
+        else {
+          fieldErrors.goalWeight = '';
+        }
+        break;
+      case 'weight':
+        if (1 > value) {
+          fieldErrors.weight = '체중을 입력하세요';
+        } else if (value > 500)
+          fieldErrors.weight = '체중은 500이하로 작성해 주세요.';
+        else {
+          fieldErrors.weight = '';
+        }
+        break;
+      case 'bodyFat':
+        if (1 > value) {
+          fieldErrors.bodyFat = '체지방량을 입력하세요.';
+        } else if (value > 100)
+          fieldErrors.bodyFat = '체지방률은 100이하로 작성해주세요.';
+        else {
+          fieldErrors.goalWeight = '';
+        }
+        break;
+      default:
+        break;
+    }
+    console.log(fieldErrors);
+    setErrors({ ...errors, ...fieldErrors });
+  };
+  return (
+    <div className="SignUp flexCenter">
+      <div className="signUpInner ">
+        <div className="logo">
+          <img src="/images/logo2.png" alt="위트 로고" />
+        </div>
+        <h1>건강 추가정보 입력</h1>
+        <p>
+          추가 정보를 입력하고 <b className="fontBold">위트만의 특별한 정보</b>
+          를 받아보세요.
+        </p>
+        <div className="inputWrapper">
+          <label>닉네임</label>
+          <div className="nickNameInput inputBox">
+            <input
+              className="nickName"
+              type="text"
+              placeholder="ex)홍길동이다"
+              name="nickname"
+              onChange={(e) => saveJoinUserInfo(e)}
+            />
+            <button>중복확인</button>
+            {errors.nickname && <div className="errors">{errors.nickname}</div>}
+          </div>
+          <div className="genderAndageBox">
+            <div className="ageInputWrapper inputBox">
+              <span>만</span>
+              <input
+                className="age"
+                type="number"
+                name="age"
+                onInput={(e) => saveJoinUserInfo(e)}
+                placeholder="ex)29"
+              />
+              {errors.age && <div className="errors">{errors.age}</div>}
+              <span>세</span>
+            </div>
```

Comment:
모든 input에 onChange 이벤트 부착할 필요 없이, 감싸고 있는 부모 요소에 한 번만 이벤트 부착해주시면 됩니다.
이벤트 위임(+ 이벤트 버블링, 이벤트 캡쳐링)이라는 개념에 관련된 부분인데, 좋은 [아티클](https://joshua1988.github.io/web-development/javascript/event-propagation-delegation/)이 있으니 꼭 읽어 보세요!
프론트엔드 면접에서 많이 물어보는 질문이기도 합니다.
```suggestion
        <div className="inputWrapper" onChange={saveJoinUserInfo}>
          <label>닉네임</label>
          <div className="nickNameInput inputBox">
            <input
              className="nickName"
              type="text"
              placeholder="ex)홍길동이다"
              name="nickname"
            />
            <button>중복확인</button>
            {errors.nickname && <div className="errors">{errors.nickname}</div>}
          </div>
          <div className="genderAndageBox">
            <div className="ageInputWrapper inputBox">
              <span>만</span>
              <input
                className="age"
                type="number"
                name="age"
                placeholder="ex)29"
              />
              {errors.age && <div className="errors">{errors.age}</div>}
              <span>세</span>
            </div>
```